### PR TITLE
[IE CLDNN] Improving prepare quantization pass

### PR DESCRIPTION
**Pull request by amolotko**

### CODE REVIEW

**Review by sshlyapn — APPROVED** [review]
Looks good to me!

**Comment by vladimir-paramuzov** [comment]
@amolotko please rebase on top of latest master

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_quantization.cpp [hunk 1]
```diff
@@ -20,415 +20,412 @@
...
+    };
+
+    auto lock_memory = [&stream] (memory::ptr memory, std::function<void(std::size_t, float)>& set_data,
+                           std::function<float(size_t)>& get_data) {
+        switch (memory->get_layout().data_type) {
+            case data_types::f32: {
+                mem_lock<float> data_lock{memory, stream};
+                float* data = data_lock.data();
+                set_data = [data] (size_t idx, float value) {
+                    data[idx] = value;
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_quantization.cpp:91]: This pointer is invalidated once `data_lock` is removed. So I believe this part must be reimplemented.

Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_quantization.cpp:91]: Fixed

**Pull request merged**